include/util/timezones.hpp

```diff
@@ -0,0 +1,44 @@
+#ifndef OSRM_TIMEZONES_HPP
+#define OSRM_TIMEZONES_HPP
+
+#include "util/log.hpp"
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/index/rtree.hpp>
+
+#include <chrono>
+
+// Time zone shape polygons loaded in R-tree
+// local_time_t is a pair of a time zone shape polygon and the corresponding local time
+// rtree_t is a lookup R-tree that maps a geographic point to an index in a local_time_t vector
+using point_t = boost::geometry::model::
+    point<double, 2, boost::geometry::cs::spherical_equatorial<boost::geometry::degree>>;
+using polygon_t = boost::geometry::model::polygon<point_t>;
+using box_t = boost::geometry::model::box<point_t>;
+using rtree_t =
+    boost::geometry::index::rtree<std::pair<box_t, size_t>, boost::geometry::index::rstar<8>>;
+using local_time_t = std::pair<polygon_t, struct tm>;
+
+std::function<struct tm(const point_t &)> LoadLocalTimesRTree(const std::string &tz_shapes_filename,
```

Comment:
This `pre-decleration` + all the `using` above (except the `point_t`) should be moved to `timezoner.cpp`